### PR TITLE
Sessions cache

### PR DESCRIPTION
**Pull request by pfcoperez**
## Description

This PR adds a local cache for built sessions within a cluster peer. Its session objects get invalidated when one of their related resources (catalog or configuration) changes across de the cluster.

### Testing
- [x] Tested by the current `HazelcastSessionProvider` tests.
- [x] Tested by the session invalidation unit tests.

### CODE REVIEW

**Comment by pfcoperez** [comment]
This PR should not be merged until https://github.com/Stratio/Crossdata/pull/610 has been merged

File: server/src/main/scala/org/apache/spark/sql/crossdata/session/HazelcastSessionProvider.scala [hunk 1]
```diff
@@ -48,6 +49,17 @@ class HazelcastSessionProvider( @transient sc: SparkContext,
   import HazelcastSessionProvider._
   import XDSharedState._
 
+  private val sessionsCache: collection.mutable.Map[SessionID, XDSession] = collection.mutable.Map.empty
```
Comment (darroyocazorla) [inline server/src/main/scala/org/apache/spark/sql/crossdata/session/HazelcastSessionProvider.scala]: The map should be thread safe

**Comment by darroyocazorla** [comment]
It would be nice to add a test ;)

**Comment by darroyocazorla** [comment]
LGTM

**Comment by pianista215** [comment]
:+1:

**Pull request merged**